meson/arch/arm32/meson.build

```diff
@@ -44,7 +44,8 @@ arch_uspace_c_args = [
 if CONFIG_FPU
 	# This is necessary for kernel too, to allow vmsr insn and fpexc manipulation.
 	# Use vfp32 to allow context save/restore of d16-d31 regs.
-	arch_uspace_c_args += [ '-mfloat-abi=hard' ]
+	# Need to use softfp because our libgcc.a uses it too
+	arch_uspace_c_args += [ '-mfloat-abi=softfp' ]
```

Comment:
I suspect this is causing the hangs on integratorcp and beaglebone (and likely also beagleboard)

Reply:
Hi, Jakub, thanks for testing! I tried recompiling (from clean state) with `hard` but I am unable to get integratorcp to boot either. It worked for you?

Reply:
Unfortunately it doesn't work for me either. Strangely though the the only two boards that are failing are the ones that are configured to use FPU via CONFIG_FPU.

Reply:
I looked into this a little more and I think the problem is that boot now uses an FPU instruction while the FPU is disabled, which results in an Undefined exception and some erratic behavior which restarts loader in a dirty way which results in executing halt() by chance.

Trace 0: 0x7fe2b406b1c0 [00000400/00003e64/00000020/ff000201] inflate_dynamic
R00=00000013 R01=00000006 R02=0000000f R03=0000011e
R04=0000c0a4 R05=0000be00 R06=0000bb28 R07=0000d3e4
R08=0000d3e4 R09=00000000 R10=0000001e R11=0000c09c
R12=00000027 R13=0000bad0 R14=0000000a R15=00003e64
PSR=600001db -ZC- A und32
Taking exception 1 [Undefined Instruction] on CPU 0
...from EL1 to EL1
...with ESR 0x0/0x2000000
Trace 0: 0x7fe2b4000280 [00000400/00000004/00000020/ff000201] 

where in inflate_dynamic there is:

    3e64:       eddf0beb        vldr    d16, [pc, #940] @ 4218 <inflate_dynamic+0x770>

I think we need to prevent boot from being built with FPU instructions.

Reply:
Changing -mfloat-abi to soft (but not softfp) makes the problem non-reproducible. The compiler is probably hesitant to make the same optimization via a call to the soft-float library. But I still think that we should look for a way to completely prevent the compiler from generating FPU instructions for loader and the kernel. As for the kernel, it's not clear to me why we have -mfpu=vfpv3 in its arch_kernel_c_args, but that's an unrelated topic.